node-graph/graph-craft/src/document.rs

```diff
@@ -180,6 +180,19 @@ impl NodeNetwork {
 			.collect();
 	}
 
+	/// Collect a hashmap of nodes with a list of the nodes that use it as input
```

Comment:
Does outward mean downstream? That's how I interpret it but I want to make sure that's correct. Maybe if you can put "(downstream/rightward)" in the comment, that'd help make it clearer. And consider, depending if you agree with me about its greater clarity, renaming the function from `outwards` to `downstream`.

Reply:
It is a map of each node to a list of the nodes that use it as input.

Reply:
This is explained in the comment linked.

Reply:
I see, I think I misinterpreted the intent by assuming something about it that wasn't relevant. Looks good to me! Ready to merge.